### DISCUSSION TITLE
Version 0.99.00 show in swagger field integer as Any

### DISCUSSION THREAD

**Discussion by ezeparziale** [fastapi/fastapi#9772] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel, Field

app = FastAPI()


class Car(BaseModel):
    id: int = Field(description="Id car", example=1)
    desc: str = Field(description="Desc car", example="NameExample")


@app.get("/")
async def root() -> Car:
    car = Car(id=1, desc="NameCar")
    return car
```

### Description

* In version 0.98.00 int fields show as integer, but in v0.99.00 show as Any in swagger.
* Something new to change?

### Operating System

Windows, macOS

### FastAPI Version

0.99.0

### Python Version

3.11.4

### Additional Context

![98](https://github.com/tiangolo/fastapi/assets/82482391/81447748-5b2e-479b-a6a7-7483de5301a9)
![99](https://github.com/tiangolo/fastapi/assets/82482391/e3e59f72-b38e-47ff-b1a4-671c589f1830)

**Comment by tiangolo** [comment, marked as answer]
That's interesting. Indeed, it shows like that, nevertheless, this is a bug in Swagger UI, not in FastAPI.

The latest version of FastAPI upgrades Swagger UI, to support OpenAPI 3.1.0, that's probably where the problem is, in Swagger UI.

But if you check the OpenAPI generated for your app, the `id` is actually correctly defined as an "integer":

```JSON
{
    "openapi": "3.1.0",
    "info": {
        "title": "FastAPI",
        "version": "0.1.0"
    },
    "paths": {
        "/": {
            "get": {
                "summary": "Root",
                "operationId": "root__get",
                "responses": {
                    "200": {
                        "description": "Successful Response",
                        "content": {
                            "application/json": {
                                "schema": {
                                    "$ref": "#/components/schemas/Car"
                                }
                            }
                        }
                    }
                }
            }
        }
    },
    "components": {
        "schemas": {
            "Car": {
                "properties": {
                    "id": {
                        "type": "integer",
                        "title": "Id",
                        "description": "Id car",
                        "example": 1
                    },
                    "desc": {
                        "type": "string",
                        "title": "Desc",
                        "description": "Desc car",
                        "example": "NameExample"
                    }
                },
                "type": "object",
                "required": [
                    "id",
                    "desc"
                ],
                "title": "Car"
            }
        }
    }
}
```

But I suspect I know where might be their error. OpenAPI 3.0.3, supported only by the previous versions of Swagger UI, was based on JSON Schema draft 00, which didn't include an `integer` type, and only had `number`. It was added in draft 06: https://json-schema.org/draft-06/json-schema-release-notes.html

And now OpenAPI 3.1.0 is based on JSON Schema 2020-12, and Swagger UI supports it, but they probably missed adding that part of displaying the "new" `integer` types correctly.

...I just went to their repo and found the issue with this specific problem: https://github.com/swagger-api/swagger-ui/issues/8959

That would be the place to track it. 🤓

**Reply by ezeparziale** [reply]
Thanks for your prompt reply, I willl track in swagger-ui. 

I migrated to v0.99.0 :rocket:
